api/client/build.go

```diff
@@ -168,8 +181,12 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 	// Dockerfile which uses trusted pulls.
 	context = replaceDockerfileTarWrapper(context, newDockerfile, relDockerfile)
 
+	// clear the progress buffer if -q is provided.
+	if buff, ok := progBuff.(*bytes.Buffer); ok {
+		buff.Reset()
```

Comment:
I was assuming that if the "downloading context from remote url" process succeeds, there is no reason to put it in stderr, if the build fails.